src/main/java/seedu/address/model/contact/NameComparator.java

```diff
@@ -10,8 +11,13 @@
      * @param s2
      * @return
      */
+
+    @Override
     public int compare(Contact s1, Contact s2) {
 
-        return s1.getName().compareTo(s2.getName());
+        Long big = Long.parseLong(s1.getName().fullName, 35);
+        Long next = Long.parseLong(s2.getName().fullName, 35);
+        return big.compareTo(next);
```

Comment:
Hi, could I check if you've tested this for long names? According to this [stackoverflow thread](https://stackoverflow.com/questions/7729886/whats-going-on-behind-the-scenes-with-method-public-static-long-parselongstri), it could fail for longer names since the return value might exceed 2^63.

Since we're comparing two strings of letters, would it be better if we just directly compared them? eg. `s1.getName().fullName > s2.getName().fullName`